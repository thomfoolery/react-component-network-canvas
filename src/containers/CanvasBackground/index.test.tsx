import React from "react";
import renderer from "react-test-renderer";
import {
  mockUseDragManager,
  mockUseGraphManager,
  mockUseWorkspace,
  mockUseBridge,
} from "@component/utils/mocks";
import {
  useDragManager,
  useGraphManager,
  useWorkspace,
  useBridge,
} from "@component/hooks";

import {CanvasBackground} from "./index";

jest.mock("@component/hooks", () => ({
  ...jest.requireActual("@component/hooks"),
  useDragManager: jest.fn(),
  useGraphManager: jest.fn(),
  useWorkspace: jest.fn(),
  useBridge: jest.fn(),
}));

useDragManager.mockImplementation(mockUseDragManager());
useGraphManager.mockImplementation(mockUseGraphManager());
useWorkspace.mockImplementation(mockUseWorkspace());
useBridge.mockImplementation(mockUseBridge());

describe("CanvasBackground", () => {
  it("renders correctly", () => {
    const transform = `translate3d(100px, 100px, 0) scale(1)`;
    const tree = renderer
      .create(<CanvasBackground transform={transform} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
