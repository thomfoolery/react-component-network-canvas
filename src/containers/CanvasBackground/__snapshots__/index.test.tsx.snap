// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasBackground renders correctly 1`] = `
<svg
  className="CanvasBackground"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    className="isDraft"
    data-testid="cc77e2e0-598e-4cb0-813c-516666955e42"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <path
      className="Edge"
      id="Edge-cc77e2e0-598e-4cb0-813c-516666955e42"
    />
    <path
      className="EdgeArea"
    />
  </g>
</svg>
`;
